metadata/src/main/java/com/google/cloud/teleport/metadata/HelperImage.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.metadata;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that marks a helper image for a Dataflow Template. The helper images will use the same
+ * underlying image as the template they are annotating.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface HelperImage {
```

Comment:
Could we split this PR into a few different PRs? It is rather large and is trying to do several things at once (refactor/cleanup, helperImage, job-builder-server(?)), and I'm having a hard time figuring out how to review it as a result.

In general, it is probably better to separate large diff refactors from other prs when possible

Reply:
Sure, I pulled out everything besides the refactoring of the Dockerfile generator. The refactor basically just consolidates all the code from the 3 separate generators into a single generator that is constructed using a builder pattern with defaults. This is also another step in the direction of allowing a single Dockerfile for all the image types

Reply:
It also adds a couple configurable options to be compatible with the HelperImage annotation in the next PR